doc/src/documentation_style_guide.rst

```diff
@@ -0,0 +1,1019 @@
+
+Then install these packages with homebrew::
+
+  brew install imagemagick graphviz docbook librsvg
+
+Install these packages with either pip or conda::
+
+  python -m pip install mpmath matplotlib sphinx sphinx-math-dollar
+
+Or::
+
+  conda install -c conda-forge mpmpath matplotlib sphinx sphinx-math-dollar
+
+**Windows 10**
+
+Making your Sphinx build successful on the Windows system is tricky because
+some dependencies like ``dvipng`` or ``docbook2x`` are not available.
+
+For Windows 10, however, the Windows Subsystem for Linux can be a possible
+workaround solution, and you can install Ubuntu shell on your Windows system
+after following the tutorial below:
+
+https://github.com/MicrosoftDocs/WSL/blob/live/WSL/install-win10.md
+
+In your command prompt, run ``ubuntu`` to transfer to Linux terminal, and
+follow the Debian/Ubuntu tutorial above to install the dependencies, and then
+you can run ``make html`` to build. (Note that you also have to install
+``make`` via ``apt-get install make``.)
+
+If you want to change the directory in your prompt to your working folder of
+SymPy in the Windows file system, you can prepend ``cd /mnt/`` to your file
+path in Windows, and run in your shell to navigate to the folder. (Also note
+that Linux uses ``/`` instead of ``\`` for file paths.)
+
+This method provides better compatibility than Cygwin or MSYS2 and more
+convenience than a virtual machine if you partially need a Linux environment
+for your workflow, however this method is only viable for Windows 10 64-bit
+users.
+
+2. Build the Documentation
+--------------------------
+
+The documentation can be built by running the ``makefile`` in the ``doc``
+subdirectory.
+
+To start, in your preferred web browser, use the drop down menu and select
+“open file” to navigate into the sympy/doc folder saved on your computer. In
+the doc folder, select the _build folder, then the html folder, and in the html
+folder, open the index.html file.
+
+To build the HTML documentation, run::
+
+  cd doc
+
+  make html
+
+This builds a local version of the documentation in ``doc/_build/html`` in your
+web browser.
+
+Open ``_build/html/index.html``.
+
+3. Make a Contribution
+----------------------
+
+For in-depth instructions on how to contribute to SymPy’s code base including
+coding conventions, creating your environment, picking an issue to fix, and
+opening a pull request, please read our full `Development Workflow
+<https://github.com/sympy/sympy/wiki/Development-workflow>`_ guide.
+
+Best Practices for Writing Documentation
+========================================
+
+Please follow these formatting, style, and tone preferences when writing
+documentation.
+
+Formatting Preferences
+----------------------
+
+**Math vs. Code**
+
+**Math:** text that is surrounded by dollar signs $ _ $ will be rendered as
+LaTeX math. Any text that is meant to appear as LaTeX math should be written as
+$math$. In the HTML version of the docs, MathJax will render the math.
```

Comment:
This is rendering incorrectly

```suggestion
``$math$``. In the HTML version of the docs, MathJax will render the math.
```

Reply:
If we are using Aaron's new extension we should link to it here.

Reply:
I'd like to see some recommendations about writing math that is suitable for reading in both the command line and via tools that render latex. I personally use IPython's `?` to display help the vast majority of time when using the SymPy documentation and there are many SymPy docstrings that are very difficult to read because of the heavy use of latex. Suggestions on using words to describe things or ASCII/Unicode based math for aspects that are important for interactive help would be nice. For example, I think recommending using unicode to display math in the text file is preferrable, if unicode offers enough capabilities. For example, you can right Pythagorean's theorem in unicode just fine and then it is readable both at the command line and through render tools. But some math equations just can't be written properly with unicode.

Reply:
By Unicode do you mean the output of `pprint()`, or something more generic? There are some problems with Unicode. One is that it requires you to format the math yourself, which can lead to inconsistencies. Another issue is that some Unicode math symbols aren't widely supported, especially on mobile devices, so you can end up with documentation that has rectangles (this is already an issue for Unicode pprint output). c.f. https://github.com/sympy/sympy/issues/15700.

Another recommendation would be to make the LaTeX as human-readable as possible. For example, if you just use the SymPy `latex()` function to generate LaTeX, it sometimes has extra stuff that isn't needed, like `\left` and `\right` commands that aren't actually required for the equation. These don't matter for the formatted equation, but they make it harder to parse from the source. Some suggestions I would say are:

- Avoid unnecessary `\left` and `\right` (but be sure to use them when they are required)
- Avoid unnecessary `{}`. e.g. write `x^2` instead of `x^{2}`. 
- Use whitespace in a way that makes the equation easiest to read. 

For example

good: `\int \sin(x)\,dx`
bad: `\int \sin{\left( x\right)}\, dx`

It would also help here if `latex()` output more human readable equations, but that would require some work to clean it up.

Reply:
> By Unicode do you mean the output of pprint()

Not necessarily. I just mean that if you can construct (in any way) a unicode version of the math you want to display without sacrificing clarity that rendered latex would offer, then using unicode in the documentation will allow it to be read via the command line and rendered versions in an equally easy way.

You raise other good points about the unicode rendering on different devices. I'm not sure that is something we can solve. Those have to be solved at the device level. But we can recommend unicode best practices to avoid some of the obvious issues on various devices.

I like your latex recommendations. That will certainly help.

Reply:
There can also be different Unicode characters for the same thing, with differing levels of support. The one thing I've seen from some math Twitter posts is if you use the normal Greek Unicode characters, it works fine, but if you instead use the same characters from the math Unicode block, they don't always have good font support. So for instance, τ (Greek Tau) usually works fine, but 𝜏 (mathematical Tau) often does not.

Reply:
I'll add these latex recommendations to this section. Thanks!